### ISSUE TITLE
[bug] False positive "initializer method does not guarantee @NonNull fields are initialized" when provably initialized because class is final

### ISSUE THREAD

**Issue by fprochazka** [uber/NullAway#323]
Hi,
I'm using version `0.7.3` and I have a problem with initializers

quoting from https://github.com/uber/NullAway/wiki/Error-Messages#initializer-method-does-not-guarantee-nonnull-field-is-initialized--nonnull-field--not-initialized

>  A field can also be initialized in a method called by the constructor, provided that the method is either private or final

Given the following class

<details>
 <summary>MutableClock.java</summary>

```java
package com.cogvio.time;

import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;

import java.time.Clock;
import java.time.Instant;
import java.time.LocalDateTime;
import java.time.ZoneId;
import java.util.*;

public final class MutableClock extends Clock
{

    private Instant instant;
    private ZoneId zone;

    private MutableClock(final Instant instant, final ZoneId zone)
    {
        this.setZone(zone);
        this.setInstant(instant);
    }

    public void setTime(final LocalDateTime time)
    {
        setInstant(Objects.requireNonNull(time, "time must not be null").atZone(zone).toInstant());
    }

    public void setInstant(final Instant instant)
    {
        this.instant = Objects.requireNonNull(instant, "instant must not be null");
    }

    public void setZone(final ZoneId zone)
    {
        this.zone = Objects.requireNonNull(zone, "zone must not be null");
    }

    @Override
    public ZoneId getZone()
    {
        return zone;
    }

    @Override
    public MutableClock withZone(final ZoneId newZone)
    {
        if (newZone.equals(this.zone)) { // intentional NPE
            return this;
        }
        return new MutableClock(instant, newZone);
    }

    @Override
    public long millis()
    {
        return instant.toEpochMilli();
    }

    @Override
    public Instant instant()
    {
        return instant;
    }

    @Override
    public String toString()
    {
        return new ToStringBuilder(this, ToStringStyle.SHORT_PREFIX_STYLE)
            .append("instant", instant)
            .append("zone", zone)
            .toString();
    }

    public static MutableClock of(final LocalDateTime time)
    {
        return of(time, ZoneId.systemDefault());
    }

    public static MutableClock of(final LocalDateTime time, final ZoneId zone)
    {
        return new MutableClock(time.atZone(zone).toInstant(), zone);
    }

}
```

</details>

NullAway reports this error:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project core-time: Compilation failure
[ERROR] .../core-time/src/main/java/com/cogvio/time/MutableClock.java:[18,13] [NullAway] initializer method does not guarantee @NonNull fields instant, zone are initialized along all control-flow paths (remember to check for exceptions or early returns).
[ERROR]     (see http://t.uber.com/nullaway )
```

and if I change the two setters to `final`, it stops complaining.

But the class itself is `final`, so there should be no reason to mark also the methods `final`.

**Comment by lazaroclapp** [comment]
Yeah, that's fair, if the class itself is final we should just treat all its methods as final. For now, consider marking the methods as `final` too as a workaround, but this should be an easy fix on our end. Will try to take a look later today or tomorrow.

**Comment by fprochazka** [comment]
@lazaroclapp thank you very much!

**Comment by lazaroclapp** [comment]
Np, @fprochazka ! FYI, we plan to cut a release with this fix early next week. Let me know if it's more urgent than that.

**Comment by fprochazka** [comment]
@lazaroclapp I've only just started annotating a huge project, and it will be at least few weeks before I can put this in CI, so no worries!

You're all doing amazing work, I was really missing a tool like this.

**Comment by lazaroclapp** [comment]
This should be fixed and available in NullAway release `0.7.5` (was also fixed in 0.7.4, but that version had a small issue with our library models for guava)

**Comment by fprochazka** [comment]
Thank you!
